python_modules/dagster/dagster/components/core/component_tree.py

```diff
@@ -348,6 +352,29 @@ def _component_decl_at_posix_path(
                 return (cp.file_path, component_decl)
         return None
 
+    def _mark_component_as_needing_reload(
```

Comment:
a bit grisly, I wonder if we need to pull out load state management beyond just the caching layer now